src/map.cpp

```diff
@@ -4672,6 +4672,28 @@ static void process_vehicle_items( vehicle *cur_veh, int part )
             apply_in_fridge(n);
         }
     }
+
+    const bool washmachine_here = cur_veh->part_flag( part, VPFLAG_WASHING_MACHINE ) && cur_veh->has_part( "WASHING_MACHINE", true );
+    bool washing_machine_finished = false;
+    if( washmachine_here ) {
+        for( auto &n : cur_veh->get_items( part ) ) {
+            const int washing_time = MINUTES( 90 );
+            int time_left = washing_time - calendar::turn.get_turn() + n.bday;
+            static const std::string filthy( "FILTHY" );
+            if( time_left == 0 ) {
+                n.item_tags.erase( filthy );
+                washing_machine_finished = true;
+                cur_veh->parts[part].enabled = false;
+            } else if( calendar::once_every( MINUTES( 15 ) ) ) {
+                add_msg( _("It should take %d minutes to finish washing items in the %s."), time_left / MINUTES( 1 ) + 1, cur_veh->name.c_str() );
+                break;
+            }
+        }
+        if( washing_machine_finished ) {
+            add_msg( _( "The washing machine in the %s has finished washing." ), cur_veh->name.c_str() );
```

Comment:
Though that would require the first item to be filthy and have bday set, meaning you'd either have to sort it to the front and require that at least one such item exists or refuse to allow washing ANY non-filthy items.

It would be much faster to process, though.